### DISCUSSION TITLE
GPU's defaulting to 0 in Timeseries forecasting

### DISCUSSION THREAD

**Discussion by alexkovaceski** [autogluon/autogluon#3863] [Q&A] [answered]
My windows machine is recognising in anaconda with AG installed that GPU=1, but when I run the forecasting model, it defaults to GPU=0.  Any idea what's going on, I am now lost for ideas.

> predictor = TimeSeriesPredictor(
...     prediction_length=48,
...     path="autogluon-m-minutes",
...     target="target",
...     eval_metric="MASE",
... freq="T",
... quantile_levels=[0.1 ,0.3, 0.7, 0.9],
... )
Warning: path already exists! This predictor may overwrite an existing predictor! path="autogluon-m-minutes"
>>>
>>> predictor.fit(
...     train_data,
...     presets="best_quality",
...     time_limit=60*120,
...     verbosity=4,
...     num_val_windows=3,
... )
Beginning AutoGluon training... Time limit = 7200s
AutoGluon will save models to 'autogluon-m-minutes'
=================== System Info ===================
AutoGluon Version:  1.0.0
Python Version:     3.10.13
Operating System:   Windows
Platform Machine:   AMD64
Platform Version:   10.0.19045
CPU Count:          12
**GPU Count:          1**
Memory Avail:       54.15 GB / 63.77 GB (84.9%)
Disk Space Avail:   637.21 GB / 1862.10 GB (34.2%)
===================================================
Setting presets to: best_quality

Fitting with arguments:
{'enable_ensemble': True,
 'eval_metric': MASE,
 'freq': 'T',
 'hyperparameters': 'default',
 'known_covariates_names': [],
 'num_val_windows': 3,
 'prediction_length': 48,
 'quantile_levels': [0.1, 0.3, 0.7, 0.9],
 'random_seed': 123,
 'refit_every_n_windows': 1,
 'refit_full': False,
 'target': 'target',
 'time_limit': 7200,
 'verbosity': 4}

train_data contains missing values represented by NaN. They have been filled by carrying forward the last valid observation.
Provided train_data has 979999 rows, 1 time series. Median time series length is 979999 (min=979999, max=979999).
Setting random seed to 123

Provided dataset contains following columns:
        target:           'target'
        past covariates:  ['aapl', 'amzn', 'googa', 'googb', 'msft', 'nvda', 'sqqq', 'tsla']
Saving autogluon-m-minutes\models\trainer.pkl
Saving autogluon-m-minutes\learner.pkl

AutoGluon will gauge predictive performance using evaluation metric: 'MASE'
        This metric's sign has been flipped to adhere to being higher_is_better. The metric score can be multiplied by -1 to get the metric value.
===================================================

Starting training. Start time is 2024-01-18 23:16:55
Saving autogluon-m-minutes\utils\data\train.pkl
Models that will be trained: ['SeasonalNaive', 'CrostonSBA', 'NPTS', 'AutoETS', 'DynamicOptimizedTheta', 'AutoARIMA', 'RecursiveTabular', 'DirectTabular', 'DeepAR', 'TemporalFusionTransformer', 'PatchTST']
Reserving 599.6s for ensemble
Training timeseries model SeasonalNaive. Training for up to 599.6s of the 7195.4s of remaining time.
        Fitting SeasonalNaive with **'num_gpus': 0,** 'num_cpus': 12
        Window 0
Shortening all time series to at most 2500
Predicting with model SeasonalNaive\W0
                -7.4844      = Validation score (-MASE)
                0.000   s    = Training runtime
                1.416   s    = Prediction runtime
        Window 1

**Comment by shchur** [comment, marked as answer]
Hi, this is a mistake in the debug logging. If you see
```
GPU Count: 1
```
at the start of AutoGluon training, then all models that support GPU (=deep learning models like TFT, PatchTST, DeepAR) will use the GPU. You can verify this by checking the GPU utilization during training.

You can safely ignore the debug message `Fitting SeasonalNaive with 'num_gpus': 0, 'num_cpus': 12`, it's logged by mistake and we will try to fix it soon.

**Reply by alexkovaceski** [reply]
Thanks
